client/termsetting/termsetting.ts

```diff
@@ -874,7 +878,19 @@ export async function fillTermWrapper(tw: TermWrapper, vocabApi: VocabApi, defau
 	await call_fillTW(tw, vocabApi, defaultQByTsHandler)
 
 	mayValidateQmode(tw)
-	return tw as TermWrapper
+}
+
+export async function fillTwLst(twlst: TermWrapper[], vocabApi: VocabApi, defaultQByTsHandler?: DefaultQByTsHandler) {
+	const ids: Array<string | undefined> = []
+	for (const tw of twlst) ids.push(tw.id)
+	const terms = await vocabApi.getTerms(ids)
+
+	for (const tw of twlst) {
+		tw.term = terms[tw.id || tw.term?.id]
+		tw.isAtomic = true
+		if (!tw.$id) tw.$id = get$id()
+		await initTermWrapper(tw, vocabApi, defaultQByTsHandler)
```

Comment:
@airenzp should `Promise.all()` be used here?

Reply:
Done

Reply:
Please see my [comment in the commit](https://github.com/stjude/proteinpaint/commit/1542863e44b464489824a5e4b7070f3cdeda651e#comments). I've fixed it in my commit.